code/game/turfs/simulated/floor_attackby.dm

```diff
@@ -173,11 +173,15 @@
 	update_icon()
 	return 1
 
+/turf/simulated/floor/try_build_cable(obj/item/stack/cable_coil/C, mob/user)
+	if(!can_build_cable(user))
```

Comment:
Looks like you're duplicating a bunch of checks by using this proc. What's the reasoning?

Reply:
The reasoning is I haven't found a way not to duplicate at least one check somewhere. And its all in the same proc, so it doesn't cause maintenance issues. And its also a very short proc. 

I could always stub out the try_build_cable() proc on /turf but then I'd copy the code three times for each /turf branches.. Or I could just build the cable without checking by default, but that might cause issues. 
I could use some suggestions tbh.

Reply:
I gave one way to avoid duplicating code in my comment above: have the check proc return a fail message if the check failed. Alternatively, the old-code variant of this pattern: have the check proc alert the user unless it's run with a silent arg.

The point is that by separating the check logic and the failure alerts into different places like this, you make it much harder to override partially.

Reply:
Okay, so I changed it and made it easier to override error messages and conditions.
I also noticed during testing that the message about the floor being too damaged is basically never reached, because the turf intercept attackby when its damaged anyways. which kind of makes me wonder why that's there now..